### PR TITLE
Add nested comment support

### LINKED ISSUE #111: Implement hierarchical structure for comments within a post
**Is your feature request related to a problem? Please describe.**
As a user, I would like to be able to see comments made on a post in a nested format, whether they be on a post I've made or a post that I've commented on.

**Describe the solution you'd like**
All comments on a post should preferably be nested so that they aren't all displayed at once. and you can reply to a specific comment.

**Describe alternatives you've considered**
To do

**Additional context**
This will be a parent issue, will be blocked until its children issues have been resolved
Blocked by #112 (implement this on the backend) and #114 (implement this on the frontend).

### LINKED ISSUE #114: Update Front-end to use new comment endpoint
**Is your feature request related to a problem? Please describe.**
Child issue of #111

#112 separates the backend to have separate routes for commenting and posting. The front end needs to be changed to use the new routes for comments.

**Describe the solution you'd like**
Front end updates their posts to use the new routes.

### PR DESCRIPTION
**Pull request by jacoballen4534**
Fixes #114 also closes #111 

## Proposed Changes

  - Update frontend to use the new comments endpoint.
  - Update backend to nest all sub-comments.
  - Update backend to allow comments to have children.
  - Update frontend to display the nested comments.

### CODE REVIEW

File: frontend/src/pages/PostDetail/PostDetailPage.jsx [hunk 1]
```diff
@@ -20,19 +20,21 @@ function formatDate(date) {
 /*
   Function used to recursively retrieve and display child comments
 */
-function nestComments(commentList, setModal) {
-  if (commentList) {
+function nestComments(commentList, setModal, setParentID) {
```
Comment (Sabatr) [inline frontend/src/pages/PostDetail/PostDetailPage.jsx:23]: This kind of function declaration isn't very consistent with the rest of the code. From what I noticed, most of them use arrows.

File: frontend/src/pages/PostDetail/PostDetailPage.jsx [hunk 2]
```diff
@@ -79,6 +86,7 @@ const PostDetailPage = ({ postToDisplay, commentsToDisplay, handleSearch, handle
         setModal={setModal}
         // eslint-disable-next-line no-underscore-dangle
```
Comment (Sabatr) [inline frontend/src/pages/PostDetail/PostDetailPage.jsx:87]: This rule seems like it conflicts with a bunch of code. If its needed and there's a few instances of such conflicts then perhaps it'd be better to overrite the rule in the `eslintrc.js` file.

File: frontend/src/pages/PostDetail/PostDetailPage.jsx (hunk 1 above)
Comment (jacoballen4534) [reply frontend/src/pages/PostDetail/PostDetailPage.jsx:23]: I have just kept the existing function declaration and updated the parameters. Should I leave it how it was before we came, or update it to make it more consistent?

Comment (Sabatr) [reply frontend/src/pages/PostDetail/PostDetailPage.jsx:23]: Oh... If you're going to make changes which are mentioned in my other comment then I'd like it changed, if not then leaving it is fine.

Comment (jacoballen4534) [reply frontend/src/pages/PostDetail/PostDetailPage.jsx:23]: Alright, I'll update the eslint rules and convert this to an arrow function.

**Pull request merged**